File: BlinkID/android/src/main/java/com/microblink/flutter/MicroblinkCreationParams.java
package com.microblink.flutter;


import java.util.Map;

public class MicroblinkCreationParams {
    OverlaySettings overlaySettings;
    String licenseKey;
    Map<String, Object> recognizerCollection;

    public MicroblinkCreationParams(Map<String, Object> args) {
        this.overlaySettings = new OverlaySettings((Map<String, Object>) args.get("overlaySettings"));
        this.licenseKey = (String) args.get("licenseKey");
        this.recognizerCollection = (Map<String, Object>) args.get("recognizerCollection");
    }
}

File: BlinkID/android/src/main/kotlin/com/microblink/blinkid/flutter/MicroblinkCreationParams.kt
package com.microblink.blinkid.flutter

@Suppress("UNCHECKED_CAST")
class MicroblinkCreationParams constructor(args: Map<String, Any?>) {
    val overlaySettings: OverlaySettings
    val licenseKey: String
    val recognizerCollection: Map<String?, Any>

    init {
        overlaySettings = OverlaySettings(
            args["overlaySettings"] as? Map<String, Any?> ?: mapOf(),
            OverlaySettings(
                useFrontCamera = false,
                flipFrontCamera = false
            )
        )
        licenseKey = args["licenseKey"] as String
        recognizerCollection = args["recognizerCollection"] as Map<String?, Any>
    }
}